### DISCUSSION TITLE
AWS provider not able to pull down my S3 images & config inconsistencies

### DISCUSSION THREAD

**Discussion by RickTheHat** [SixLabors/ImageSharp.Web#297] [General]
I was playing around with this on .NET 6 and of course, didn't read the supported frameworks until a couple hours into debugging so glad I finally noticed. Is upgrading this to 6 or 7 on the roadmap? Looks like .NET Core 3.1 just went out of support in Dec 2022 - [https://en.wikipedia.org/wiki/.NET](https://en.wikipedia.org/wiki/.NET)

**Comment by JimBobSquarePants** [comment]
The target framework doesn’t require updating. Anything targeting .NET Core 3.1 will automatically run on a .NET 6 install.

**Comment by RickTheHat** [comment]
tx for the quick response and I will update my issue title then b/c trying to pull my S3 images down but it's not working and no good error messages.

Turns out I'm actually using an MVC Core .NET 7 web project and trying to use **SixLabors.ImageSharp.Web.Providers.AWS** to which I found some inconsistencies with the documentation compared to what I had to do in order to remove the red squigly lines and get it to compile, see both pics below:

![CapturFiles-20230102_113934](https://user-images.githubusercontent.com/704833/210305476-236a18d5-6d58-428f-8f02-aa90b9daa27a.jpg)

![CapturFiles-20230102_113722](https://user-images.githubusercontent.com/704833/210305489-1eea8520-a9dc-46c6-936c-4fb00fc0de16.jpg)

The only way I can get the red squigly line to go away is to use AWSS3StorageImageProvider without the "Options" 
![CapturFiles-20230102_115420](https://user-images.githubusercontent.com/704833/210306146-bbef6207-d713-419c-97be-2f152ff74f8e.jpg)
which allows me to compile and run the project however it's not recognizing my S3 bucket and get a 404 when trying to pull down images and no errors unfortunately.

If anyone else has run into this with .NET 7, lmk what you found to fix this, thanks in advance!

PS I am able to upload files to my S3 bucket so I am connecting with it correctly and this package is asking for the same things (bucket name, key, secret, region etc.) just not pulling down my images.

**Reply by JimBobSquarePants** [reply]
See https://github.com/SixLabors/docs/pull/55 (just merged)

You should drop the `XXOptions` suffix in the generic parameter.
